agent/taskresource/cgroup.go

```diff
@@ -0,0 +1,225 @@
+// +build linux
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package taskresource
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+	"github.com/cihub/seelog"
+	"github.com/containerd/cgroups"
+	"github.com/pkg/errors"
+)
+
+const (
+	memorySubsystem         = "/memory"
+	memoryUseHierarchy      = "memory.use_hierarchy"
+	rootReadOnlyPermissions = os.FileMode(400)
+)
+
+var (
+	enableMemoryHierarchy = []byte(strconv.Itoa(1))
+)
+
+// CgroupResource represents Cgroup resource
+type CgroupResource struct {
```

Comment:
What's the intended difference between taskresource/cgroupresource and resources/cgroup? why are we using another layer of abstraction? Can we combine?

Reply:
Sure. I left the package `resources/cgroup` as is, since the current flow of cgroup creation/deletion in engine is dependent on it. Also in `taskresource/cgroup.go`, I am just reusing the methods and mocks under `resources/cgroup`. Once the cgroup create/cleanup methods in `taskresource/cgroup.go` are hooked into engine, the plan is to move everything under `resources/cgroup` to `taskresource` package.